src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -219,6 +220,8 @@ BuildSelectStatementViaStdPlanner(Query *masterQuery, List *masterTargetList,
 	 * citus_extradata_container function call in the masterQuery).
 	 */
 	remoteScan->custom_scan_tlist = copyObject(masterTargetList);
+	if (InsertSelectIntoLocalTable(masterQuery))
+		((Var *)((TargetEntry *)lfirst(remoteScan->custom_scan_tlist->head))->expr)->varno = 3;
```

Comment:
what does this do?

Reply:
This is a temporary fix until I find out why changing the masterQuery makes planning break the targetlists in subplan. I was getting `variable not found in subplan target list` error.

Reply:
This is I think the reason of the error [here](https://github.com/citusdata/citus/pull/3870#issuecomment-639367380).